packages/ai-jsx/src/batteries/docs.tsx

```diff
@@ -552,17 +681,22 @@ export interface DocsQAProps {
    *  }
    * ```
    */
-  chunkFormatter: (props: { doc: ScoredChunk }) => Node;
+  chunkFormatter: (props: { doc: ScoredChunk<ChunkMetadata> }) => Node;
 }
 /**
  * A component that can be used to answer questions about documents. This is a very common usecase for LLMs.
  */
-export async function DocsQA(props: DocsQAProps) {
-  const status = (await props.corpus.getStats()).loadingState;
-  if (status !== Corpus.LoadingState.COMPLETED) {
-    return `Corpus is not loaded. It's in state: ${status.toString()}`;
+export async function DocsQA<ChunkMetadata extends Jsonifiable = Jsonifiable>(props: DocsQAProps<ChunkMetadata>) {
+  let docs: ScoredChunk<ChunkMetadata>[];
+  try {
+    docs = await props.corpus.search(props.question, { limit: props.chunkLimit });
+  } catch (e) {
+    if (e instanceof CorpusNotReadyError) {
+      return e.message;
```

Comment:
I'm slightly inclined to say we should just throw the error and force the user to handle it. They can use an `ErrorBoundary` if they want.

Reply:
sg, done.